.github/workflows/build-and-test.yml

```
@@ -0,0 +1,51 @@
+name: facebook/metro/build-and-test
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+# head_ref is per PR, so this ensures that updating the latest PR commit
+# will cancel the previous run of the workflow and trigger a new one
+concurrency:
+  group: "build-and-test-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  run-js-checks:
+    runs-on: ubuntu-latest
+    name: "Typescript, Lint, Smoke Test"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/yarn-install
+      - run: yarn typecheck
+      - run: yarn typecheck-ts
+      - run: yarn lint
+      - run: yarn test-smoke
+
+  test-with-coverage:
+    runs-on: ubuntu-latest
+    name: "Test Coverage Calculation"
```

nit (all tests need to run to calculate coverage)
```suggestion
    name: "Tests with coverage"
```